### ISSUE TITLE
I think with Serializer you should also call get_fields()

### ISSUE THREAD

**Issue by rrauenza** [adenh93/django-typomatic#12]
In your code, you go after `_declared_fields`, but the Serializer implementation actually implements `get_fields()`.  Bypassing this method makes it difficult to derive my own Serializer that builds its own fields.

https://github.com/adenh93/django-typomatic/blob/bd316ddf635f80406d44262b29758e1933b171bf/django_typomatic/__init__.py#L103

https://github.com/encode/django-rest-framework/blob/0407a0df8a16fdac94bbd08d49143a74a88001cd/rest_framework/serializers.py#L377

In other words, I think these two cases are the same and should just do the ModelSerializer method?:

```
    if issubclass(serializer, serializers.ModelSerializer):
        instance = serializer()
        fields = instance.get_fields().items()
    else:
        fields = serializer._declared_fields.items()
```

**Comment by rrauenza** [comment]
https://github.com/adenh93/django-typomatic/pull/13

**Comment by adenh93** [comment]
Approved the PR and merged 👍
